src/main/scala/org/constellation/genesis/Genesis.scala

```diff
@@ -23,156 +25,144 @@ object Genesis extends StrictLogging {
-      dao.checkpointService.put(initialBlock2) >>
-      IO(dao.recentBlockTracker.put(genesisBlock)) >>
-      IO(dao.recentBlockTracker.put(initialBlock1)) >>
-      IO(
-        dao.recentBlockTracker
-          .put(initialBlock2)
-      ))
-    // TODO: Get rid of unsafeRunSync
-      .unsafeRunSync()
-
-    go.genesis.transactions.foreach { rtx =>
-      val bal = rtx.amount
-      dao.addressService
-        .set(rtx.dst.hash, AddressCacheData(bal, bal, Some(1000d), balanceByLatestSnapshot = bal))
-        // TODO: Get rid of unsafeRunSync
-        .unsafeRunSync()
-    }
-
-    dao.genesisObservation = Some(go)
-    dao.genesisBlock = Some(go.genesis)
-
-    /*
-    // Dumb way to set these as active tips, won't pass a double validation but no big deal.
-    checkpointMemPool(go.initialDistribution.baseHash) = go.initialDistribution
-    checkpointMemPool(go.initialDistribution2.baseHash) = go.initialDistribution2
-    checkpointMemPoolThresholdMet(go.initialDistribution.baseHash) = go.initialDistribution -> 0
-    checkpointMemPoolThresholdMet(go.initialDistribution2.baseHash) = go.initialDistribution2 -> 0
+    for {
+      _ <- Sync[F].delay(logger.debug("Started genesis acceptance!"))
+      genesisBlock = CheckpointCache(go.genesis, height = Some(Height(0, 0)))
+      initialBlock1 = CheckpointCache(go.initialDistribution, height = Some(Height(1, 1)))
+      initialBlock2 = CheckpointCache(go.initialDistribution2, height = Some(Height(1, 1)))
+      _ <- LiftIO[F].liftIO(dao.soeService.put(go.genesis.soeHash, go.genesis.soe))
+      _ <- LiftIO[F].liftIO(dao.soeService.put(go.initialDistribution.soeHash, go.initialDistribution.soe))
+      _ <- LiftIO[F].liftIO(dao.soeService.put(go.initialDistribution2.soeHash, go.initialDistribution2.soe))
+      _ <- LiftIO[F].liftIO(dao.checkpointService.put(genesisBlock))
+      _ <- LiftIO[F].liftIO(dao.checkpointService.put(initialBlock1))
+      _ <- LiftIO[F].liftIO(dao.checkpointService.put(initialBlock2))
+      _ <- Sync[F].delay(dao.recentBlockTracker.put(genesisBlock))
+      _ <- Sync[F].delay(dao.recentBlockTracker.put(initialBlock1))
+      _ <- Sync[F].delay(dao.recentBlockTracker.put(initialBlock2))
+      // Changed it from setting balance one by one for every tx to setting them all at once with setAll
+      genesisBalances = go.genesis.transactions.map { rtx =>
+        val bal = rtx.amount
+        rtx.dst.hash -> AddressCacheData(bal, bal, Some(1000d), balanceByLatestSnapshot = bal)
+      }.toMap
+      _ <- LiftIO[F].liftIO(dao.addressService.setAll(genesisBalances))
+      _ <- Sync[F].delay {
+        dao.genesisObservation = Some(go)
+        dao.genesisBlock = Some(go.genesis)
+      }
+      _ <- dao.metrics.updateMetricAsync("genesisAccepted", "true")
+      _ <- if (setAsTips)
+        List(go.initialDistribution, go.initialDistribution2)
+          .traverse(cb => LiftIO[F].liftIO(dao.concurrentTipService.update(cb, Height(1, 1), isGenesis = true)))
```

Comment:
please add an error handling when using traverse. if one of these updates fail, the whole genesis flow fails